flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/PromptHandler.java

```diff
@@ -0,0 +1,188 @@
+                            i++;
+                            break;
+                        case 'c':
+                            promptStringBuilder.append(executor.getCurrentCatalogName(sessionId));
+                            i++;
+                            break;
+                        case 'd':
+                            promptStringBuilder.append(executor.getCurrentDatabaseName(sessionId));
+                            i++;
+                            break;
+                        case ESCAPE_BACKSLASH:
+                            promptStringBuilder.append(ESCAPE_BACKSLASH);
+                            i++;
+                            break;
+                            // date time pattern \{...\}
+                        case '{':
+                            int dateTimeMaskCloseIndex =
+                                    pattern.indexOf(ESCAPE_BACKSLASH + "}", i + 1);
+                            if (dateTimeMaskCloseIndex > 0) {
+                                String mask = pattern.substring(i + 2, dateTimeMaskCloseIndex);
+                                promptStringBuilder.append(
+                                        FORMATTER_CACHE.get(mask).format(new Date()));
+                                i = dateTimeMaskCloseIndex + 1;
+                            }
+                            break;
+                            // color and style pattern \[...\]
+                        case '[':
+                            int closeBracketIndex = pattern.indexOf(ESCAPE_BACKSLASH + "]", i + 1);
+                            if (closeBracketIndex > 0) {
+                                String color = pattern.substring(i + 2, closeBracketIndex);
+                                AttributedStyle style = STYLE_RESOLVER.resolve(color);
+                                promptStringBuilder.style(style);
+                                i = closeBracketIndex + 1;
+                            }
+                            break;
+                            // property value pattern \:...\:
+                        case ':':
+                            int nextColonIndex = pattern.indexOf(ESCAPE_BACKSLASH + ":", i + 1);
+                            String propertyValue;
+                            if (nextColonIndex > 0) {
+                                String propertyName = pattern.substring(i + 2, nextColonIndex);
+                                propertyValue =
+                                        executor.getSessionConfigMap(sessionId).get(propertyName);
+                                promptStringBuilder.append(propertyValue);
+                                i = nextColonIndex + 1;
+                            }
+                            break;
+                    }
+                } else {
+                    promptStringBuilder.append(c);
+                }
+            }
+            return promptStringBuilder.toAnsi();
+        } catch (IllegalArgumentException e) {
+            terminalSupplier
```

Comment:
yes, sure
I've added a test for that https://github.com/apache/flink/pull/17830/files#diff-464a0c9ad1840825e48db9f3c6039dd62e58e5bb5061e28f9dc89f86793ff60cR178
In case of `Exception` a default value for prompt will be applied (`SqlClientOptions.PROMPT.defaultValue()`)